src/ui/overworld/OverworldHUD.tsx

```
@@ -0,0 +1,166 @@
+  const [showSC, setShowSC] = useState(false);
+  const { audioState: isPlaying, handleAudioChange } = useAudioControl(
+    useCallback((state: unknown) => {
+      if (typeof state !== 'object' || state === null) return false;
+      const audioState = state as { currentSongId?: unknown; isPlaying?: unknown };
+      return audioState.currentSongId === 'ambient' && audioState.isPlaying === true;
+    }, []),
+    { pollEvenWithSubscribe: true, pollMs: 1000 }
+  );
+
+  const isMuted = muted ?? !isPlaying;
+  const handleToggleMute = onToggleMute ?? useCallback(() => {
```

Biome (2.4.12)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

(lint/correctness/useHookAtTopLevel)